src/linker/utilities/singularity_utils.py

```diff
@@ -9,28 +9,14 @@ def run_with_singularity(
     input_data: List[Path],
     results_dir: Path,
     container_path: Path,
-    step_name: str,
-    implementation_name: str,
 ) -> None:
     logger.info("Running container with singularity")
-    # TODO [MIC-4708]: break this singularity implementation_dir hard-coding
-    ## Need to figure out how to add files to singualrity container rather than
-    ## changing the pwd of the container
-    implementation_dir = (
-        Path(__file__).parent.parent
-        / "steps"
-        / step_name
-        / "implementations"
-        / implementation_name
-    )
-    _run_container(input_data, results_dir, implementation_dir, container_path)
-    _clean(results_dir, implementation_dir, container_path)
+    _run_container(input_data, results_dir, container_path)
+    _clean(results_dir, container_path)
 
 
-def _run_container(
-    input_data: List[Path], results_dir: Path, implementation_dir: Path, container_path: Path
-) -> None:
-    cmd = f"singularity run --pwd {implementation_dir} --bind /tmp:/tmp --bind {results_dir}:/results "
+def _run_container(input_data: List[Path], results_dir: Path, container_path: Path) -> None:
+    cmd = f"singularity run --containall --no-home --bind /tmp:/tmp --bind {results_dir}:/results "
```

Comment:
This is the single most important line in this PR. We are no longer setting the pwd and instead relying on the default working directory ('/'). Further, --containall and --no-home limit what gets mounted when we run the container. Note that this does mean we need to re-build all existing containers with the implementation scripts saving results to /results/ (currently they're just saving to the working directory).